packages/sharing-smart-contract/contracts/DataProtectorSharing.sol

```diff
@@ -166,7 +178,15 @@ contract DataProtectorSharing is
             _workerpoolOrder.category
         );
 
-        dealid = POCO_DELEGATE.matchOrders(_appOrder, _datasetOrder, _workerpoolOrder, requestOrder);
+        if (_useVoucher) {
+            IVoucher _voucher = IVoucher(VOUCHER_HUB.getVoucher(_spender));
+            dealid = _voucher.matchOrders(_appOrder, _datasetOrder, _workerpoolOrder, requestOrder);
+        } else {
+            if (_workerpoolOrder.workerpoolprice > 0) {
+                POCO_DELEGATE.transferFrom(_spender, address(this), _workerpoolOrder.workerpoolprice);
```

Comment:
we may want to keep the track of the value transferred associated with dealid to enable later refund in case of claim on a failing poco task

```suggestion
                // TODO: keep the track of value transferred for deal/task refund?
                POCO_DELEGATE.transferFrom(_spender, address(this), _workerpoolOrder.workerpoolprice);
```